### ISSUE TITLE
Unicode width not being accounted for

### ISSUE THREAD

**Issue by nithinmuthukumar** [zhiburt/tabled#349]
Characters with variable width break the table borders.
Example:
![Screen Shot 2023-06-28 at 12 18 23 PM](https://github.com/zhiburt/tabled/assets/33675303/2f0e4a20-a43f-4348-a2f0-e56952605fbe)

**Comment by zhiburt** [comment]
Hi @nithinmuthukumar 

Could you provide your example data?

**Comment by daywalker90** [comment]
I'm encountering the same problem. What does the color feature have to do with this? Is there something i can do, besides filtering unicode characters?

**Comment by daywalker90** [comment]
It's not all characters but these 2 i found causing issues: ☀️🏛️

**Comment by zhiburt** [comment]
Hi @daywalker90 

Could you provide a code example?

> What does the color feature have to do with this?

It was renamed to  `ansi` in recent releases.
But it changes the calculations a little bit.
But to be honest in this specific case, it mustn't be different.

> It's not all characters but these 2 i found causing issues: ☀️🏛️

Generally speaking it all depends on a terminal.
If you render the same table, which includes some obscure utf8, in different terminals it **may** (likely) be different.

**Comment by zhiburt** [comment]
> Generally speaking it all depends on a terminal.
If you render the same table, which includes some obscure utf8, in different terminals it may (likely) be different.

Now I am thinking maybe it's worth to add a configuration for it.
I mean we can't fetch information from a terminal,
But we could be given it.

Either a complete UTF-8 map or some specifics like *this particular char will take not 1 but 3 terminal columns*.
Yee I could could be usefullllllllllllllllllll rarely, but would slow down rendering a fair bit.
So maybe would need to be hidden by a feature.

Just thoughts here.

**Comment by daywalker90** [comment]
You are right, it's highly terminal related. With [putty](https://www.chiark.greenend.org.uk/~sgtatham/putty/) i have these 2 above misbehaving and with powershell it's a complete mess. (i know, windows... yikes)
```rust
pub async fn qwert(
    plugin: Plugin<()>,
    _args: serde_json::Value,
) -> Result<serde_json::Value, Error> {
    let row1 = TableTest {
        name: "test1".to_string(),
    };
    let row2 = TableTest {
        name: "test☀️".to_string(),
    };
    let mut rows = Vec::new();
    rows.push(row1);
    rows.push(row2);
    let table = Table::new(rows);
    Ok(json!({ "format-hint":"simple",
        "result": table.to_string() }))
}
```

**Comment by zhiburt** [comment]
See `xterm`

![image](https://github.com/zhiburt/tabled/assets/20165848/cb735017-d3c5-4118-bb2f-a78150581c30)
